processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java

```diff
@@ -165,8 +168,12 @@ public boolean execute(List<RawResultIterator> resultIteratorList) {
         readAndLoadDataFromSortTempFiles();
       }
       isCompactionSuccess = true;
-    } catch (Exception e) {
+    } catch (BadRecordFoundException e) {
```

Comment:
BadRecordFoundException  will never come in case of compaction so remove this catch block